app/models/protected_area.rb

```diff
@@ -90,23 +90,23 @@ def is_green_list
   def self.greenlist_coverage_growth(start_year = nil)
     # Is in this format: {year: area, ...}
     # Takes an optional start year from which to start counting
-    coverage_growth_hash = {}
-
     areas = ProtectedArea.non_candidate_green_list_areas.where.not(legal_status_updated_at: nil)
     
+    sorted_dates = areas.pluck(:legal_status_updated_at).sort { |a,b| b <=> a }.uniq
+
     if start_year
       date_from_year = "#{start_year}-01-01 00:00:00".to_time 
-      areas = areas.where("legal_status_updated_at >= ? ", date_from_year)
+      sorted_dates.filter! { |date| date >= date_from_year }
     end
```

Comment:
I would do all of this with just one SQL/ActiveRecord query for better performance and readability.
Both filtering (including non candidate sites) and sorting can be done within the same query.

Ideally, the entire process of cumulative sum could be achieved with a single query;  that's a little more complicated but feel free to give it a go if you like :)